firmware/robot2015/src-ctrl/modules/control/PidMotionController.hpp

```diff
@@ -0,0 +1,83 @@
+#pragma once
+
+#include <array>
+#include "Pid.hpp"
+#include "RobotModel.hpp"
+
+/**
+ * Robot controller that runs a PID loop on each of the four wheels.
+ */
+class PidMotionController {
+public:
+    PidMotionController() {
+        setPidValues(1, 0, 0);
+
+        for (auto& ctrl : _controllers) {
+            ctrl.setWindup(5);
+        }
+    }
+
+    void setPidValues(float p, float i, float d) {
+        for (Pid& ctl : _controllers) {
+            ctl.kp = p;
+            ctl.ki = i;
+            ctl.kd = d;
+        }
+    }
+
+    void setTargetVel(Eigen::Vector3f target) { _targetVel = target; }
```

Comment:
I think in the case a move is equivalent to a copy since the values in the vector are allocated directly in the object, rather than on the heap.